android/src/main/java/com/doublesymmetry/trackplayer/service/MusicService.kt

```diff
@@ -258,8 +228,9 @@ class MusicService : HeadlessJsTaskService() {
     }
 
     @MainThread
-    fun stop() {
+    fun stopPlayer() {
         player.stop()
+        player.destroy()
```

Comment:
This function should only be called in `onDestroy` of the service, but I see we also call it in `reset` react method. Do we need reset at this point? It's a very ambiguous function too, since we don’t really know what exactly we want to reset, but we are calling random functions in both ios and android in it.

in case of Android, it will break the service since we are indeed destroying the player with no way of recreating it. It's supposed to never get destroyed until the service dies:

```kotlin
  @ReactMethod
    fun reset(callback: Promise) = scope.launch {
        if (verifyServiceBoundOrReject(callback)) return@launch

        musicService.stopPlayer()
        callback.resolve(null)
    }
```